cmd/server/handler.go

```diff
@@ -154,6 +155,21 @@ func RunServeAll(version, build, date string) func(cmd *cobra.Command, args []st
 	}
 }
 
+func setTracingLogger(d driver.Driver, logger *negronilogrus.Middleware) {
+	logger.Before = func(entry *logrus.Entry, r *http.Request, remoteAddr string) *logrus.Entry {
```

Comment:
This overwrites the existing logger.Before, right? I think we have a custom Before somewhere - Maybe you could check if `logger.Before` is set and if it is, execute it and then execute this logic here, but only adding the `trace_id` and `span_id`?

Reply:
I see, it appears that we're not actually using that here. Here's the actual function I was referring to: https://github.com/ory/x/blob/master/reqlog/middleware.go

I think it would make sense to use that function and add the trace_id in that middleware. We should also add it to both before, and after!

Reply:
Here's an example: https://github.com/ory/oathkeeper/blob/master/cmd/server/server.go#L44

Reply:
I fixed it!
It seemed `logger.Before` was set default(`request`, `method` and `remote`). I realized it's not safety to consider whether `logger.Before` or not, I fixed to add field.

> We should also add it to both before, and after!

It's seems it's enough to add only before. Actually logs appear `span_id` and `trace_id` at both stating and completed handling request.

```
{"level":"info","method":"GET","msg":"started handling request","remote":"[::1]:56308","request":"/health/alive","time":"2020-01-11T11:09:46+09:00","trace_id":"00f067aa0ba902b7"}
{"level":"info","measure#hydra/admin: https://localhost:4444/.latency":218780,"method":"GET","msg":"completed handling request","remote":"[::1]:56308","request":"/health/alive","status":200,"text_status":"OK","time":"2020-01-11T11:09:46+09:00","took":218780,"trace_id":"00f067aa0ba902b7"}
```